tests/Controllers/UserControllerTest.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace Tests\Controllers;
+
+use App\Models\Project;
+use App\Models\Role;
+use App\Models\User;
+use App\Repositories\UserRepository;
+use Illuminate\Foundation\Testing\DatabaseTransactions;
+use Mockery\MockInterface;
+use Tests\TestCase;
+
+class UserControllerTest extends TestCase
+{
+    use DatabaseTransactions;
+
+    /** @var MockInterface */
+    protected $userRepo;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+        $this->signInWithDefaultAdminUser();
+        $this->withHeaders(['X-Requested-With' => 'XMLHttpRequest']);
+    }
+
+    private function mockRepository()
+    {
+        $this->userRepo = \Mockery::mock(UserRepository::class);
+        app()->instance(UserRepository::class, $this->userRepo);
+    }
+
+    public function tearDown(): void
+    {
+        parent::tearDown();
+        \Mockery::close();
+    }
+
+    /** @test */
+    public function test_can_retrieve_user_with_its_project_and_roles()
+    {
+        factory(User::class)->create();
```

Comment:
always define variable no matter its used or not, and use real name for user everywhere